synthtool/gcp/common.py

```diff
@@ -225,17 +225,9 @@ def py_library(self, **kwargs) -> Path:
             kwargs["default_python_version"] = "3.8"
         if "unit_test_python_versions" not in kwargs:
             kwargs["unit_test_python_versions"] = ["3.6", "3.7", "3.8", "3.9"]
-            if "microgenerator" not in kwargs:
```

Comment:
After this change, `microgenerator` is only used [here](https://github.com/googleapis/synthtool/blob/5f330a45e76ad2b3551c5276aaa818de324632b5/synthtool/gcp/templates/python_library/noxfile.py.j2). Could we remove `microgenerator` altogether?

Reply:
The following repos do *not* have the `asyncmock` / `pytest-asyncio` test dependencies installed in their `noxfile.py`:

```bash
$ for noxfile in python-*/noxfile.py; do grep -q asyncmock $noxfile || echo $noxfile; done
python-bigquery/noxfile.py
python-bigquery-sqlalchemy/noxfile.py
python-cloud-core/noxfile.py
python-crc32c/noxfile.py
python-irm/noxfile.py
python-ndb/noxfile.py
python-spanner-django/noxfile.py
python-storage/noxfile.py
python-test-utils/noxfile.py
```

Of those these repos:

- `python-bigquery`: `owlbot.py` excludes `noxfile.py` from copying
- `python-bigquery-sqlalchemy`: pins explicit `{unit,system}_test_python_versions`, has no `async` support
- `python-cloud-core`: `owlbot.py` excludes `noxfile.py` from copying
- `python-crc32c`: has no `owlbot.py` (or `synth.py`)
- `python-irm`: repo is abandoned
- `python-ndb`: uses `templated_files` only for `.kokoro/`
- `python-spanner-django`: uses `templated_files` only for `.kokoro/` / `.trampolinerc`
- `python-storage`: `owlbot.py` excludes `noxfile.py` from copying
- `python-test-utils`: `owlbot.py` excludes `noxfile.py` from copying

Reply:
@tseaver My initial thought is we should be ok removing `microgenerator`. If we encounter any issue, we can add repo specific customizations in owlbot.py. WDYT?

Reply:
39e8fc7